### PR TITLE
1.21 Support

### PR DESCRIPTION
**Pull request by JongWasTaken**
I know this mod isn't being updated anymore, but I _really_ like it, so I did it myself:  

This PR adapts the existing code to Minecraft 1.21.  
  
Quick overview of all changes:  
- Update Loom, Fabric and Java
- `new Identifier()` changed to `Identifier.of()`
- BlockScreenOpener uses components instead of NBT now, greatly simplifying code
- `BlockView` was renamed to `WorldView`
- Removed some `@Overwrite`'s in `FakeWorld`, because they don't exist in `ServerWorld` anymore
  
Tested with Shulker Box, Anvil (all types), Enchantment Table and Crafting Table.  
  
These changes could also easily be adapted to work with 1.20.5 if necessary.  
  
(Also, you probably want to create a new 1.21 branch for this instead of merging it into 1.20.1)

### CODE REVIEW

**Comment by manuelgrabowski** [comment]
Thank you so much for sharing this, @JongWasTaken! One of the most essential mods for me ever since I started using it in 1.19 – it basically solves the inventory problem, at least in my book. I was looking into upgrading this myself a few weeks back, but that was my first deeper look into MC modding in a while, so I ended up getting stuck quickly. So I worked on two simpler mods the last two weeks instead, and now just came back here to give it another go, but finding you have already done it! 🎉 

Your version is running great for me! Do you plan putting up a fork on Modrinth? If not, I might even consider doing so myself… I want more people to get this amazing experience. 😄

**Comment by JongWasTaken** [comment]
Thank you for the kind words @manuelgrabowski!  
I also much prefer this mod over similar backpack type of mods, as it finally gives shulkers more of a reason to be used. I guess some people would disagree because of the sheer amount of inventory size you can get with this, but as the owner of a build-centric survival server, this mod is an absolute godsend when it comes to having building resources on hand.  
  
Concerning publishing a fork, I am a bit reluctant, as I don't want to claim @Hoid2's work without express permission to do so first. I think the ideal solution would be for them to merge PRs like this and to continue publishing releases, though I also understand if they don't want to do so.  
So basically, if Hoid2 gives me permission, I would be more than happy to continue maintaining a fork of this and publishing it on Modrinth and CurseForge myself. Otherwise I will just continue to post PRs right here.  
  
Either way, I wish everybody reading this a great day (or rest of it)!

**Comment by manuelgrabowski** [comment]
Ah, I realize now that I had mixed up this situation with another mod that hasn't been updated yet… still waiting on a bunch more, the 1.20.5 NBT/component change understandably killed off quite a few mods' further development it seems.

On the one I was thinking about when writing my earlier comment, the original dev has stopped engaging with the project for over a year now, even befoe the 1.20.5 changes… so it seems unlikely they'd come back to give or deny permission. In a situation like that I think it's fair to just follow what the license allows. But totally with you in this case, @Hoid2 should definitely have a chance to consider just merging this themselves. If you're reading this, Hoid – sorry if I sounded overly dismissive. I'm so grateful for how much more awesome you made the last few years of playing Minecraft for me!

What I love about the approach of this mod is also how "vanilla" it is – no new special items for more storage, just making what's already in the game so much more accessible. Because of that it's also wonderfully "backwards-compatible", no need to worry about what happens to my items if I stop using the mod. For me it's not even so much about the inventory space that this enables, but the ease of use. Constantly placing my ender chest, and then shulker boxes only to take out a new stack of rockets or similar, and then breaking both containers again… I've done that long enough, that's just not doing anything for me gameplay-wise. I'd rather use my time in MC doing actual things than repeating these steps all the time. 😄

**Comment by Hoid2** [comment]
@JongWasTaken if you would like to maintain/publish a fork you have my
permission. I'm sure many people would appreciate that.
I'm not in a position to do any development (including merging prs) and
won't be for a couple of years.
